cmd/snap-seccomp/main_test.go

```diff
@@ -0,0 +1,519 @@
+		{"prctl PR_CAP_AMBIENT_RAISE", "prctl;native;99", main.SeccompRetKill},
+
+		// test_bad_seccomp_filter_args_prio
+		{"setpriority PRIO_PROCESS 0 >=0", "setpriority;native;PRIO_PROCESS,0,19", main.SeccompRetAllow},
+		{"setpriority PRIO_PROCESS 0 >=0", "setpriority;native;99", main.SeccompRetKill},
+
+		// test_bad_seccomp_filter_args_quotactl
+		{"quotactl Q_GETQUOTA", "quotactl;native;Q_GETQUOTA", main.SeccompRetAllow},
+		{"quotactl Q_GETQUOTA", "quotactl;native;99", main.SeccompRetKill},
+
+		// test_bad_seccomp_filter_args_socket
+		{"socket AF_UNIX", "socket;native;AF_UNIX", main.SeccompRetAllow},
+		{"socket AF_UNIX", "socket;native;99", main.SeccompRetKill},
+		{"socket - SOCK_STREAM", "socket;native;-,SOCK_STREAM", main.SeccompRetAllow},
+		{"socket - SOCK_STREAM", "socket;native;-,99", main.SeccompRetKill},
+
+		// test_bad_seccomp_filter_args_termios
+		{"ioctl - TIOCSTI", "ioctl;native;-,TIOCSTI", main.SeccompRetAllow},
+		{"ioctl - TIOCSTI", "ioctl;native;-,99", main.SeccompRetKill},
+	} {
+		simulateBpf(c, t.seccompWhitelist, t.bpfInput, t.expected)
+	}
+}
+
+func (s *snapSeccompSuite) TestCompileBadInput(c *C) {
+	for _, t := range []struct {
+		inp    string
+		errMsg string
+	}{
+		// test_bad_seccomp_filter_args_clone (various typos in input)
+		{"setns - CLONE_NEWNE", `cannot parse line: cannot parse token "CLONE_NEWNE" \(line "setns - CLONE_NEWNE"\)`},
+		{"setns - CLONE_NEWNETT", `cannot parse line: cannot parse token "CLONE_NEWNETT" \(line "setns - CLONE_NEWNETT"\)`},
+		{"setns - CL0NE_NEWNET", `cannot parse line: cannot parse token "CL0NE_NEWNET" \(line "setns - CL0NE_NEWNET"\)`},
+
+		// test_bad_seccomp_filter_args_mknod (various typos in input)
+		{"mknod - |S_IFIF", `cannot parse line: cannot parse token "S_IFIF" \(line "mknod - |S_IFIF"\)`},
+		{"mknod - |S_IFIFOO", `cannot parse line: cannot parse token "S_IFIFOO" \(line "mknod - |S_IFIFOO"\)`},
+		{"mknod - |S_!FIFO", `cannot parse line: cannot parse token "S_IFIFO" \(line "mknod - |S_!FIFO"\)`},
+
+		// test_bad_seccomp_filter_args_null
+		{"socket S\x00CK_STREAM", `cannot parse line: cannot parse token .*`},
+		{"socket SOCK_STREAM\x00bad stuff", `cannot parse line: cannot parse token .*`},
+
+		// test_bad_seccomp_filter_args
+		{"setpriority bar", `cannot parse line: cannot parse token "bar" .*`},
+		{"setpriority -1", `cannot parse line: cannot parse token "-1" .*`},
+		{"setpriority 0 - -1 0", `cannot parse line: cannot parse token "-1" .*`},
+		{"setpriority --10", `cannot parse line: cannot parse token "--10" .*`},
+		{"setpriority 0:10", `cannot parse line: cannot parse token "0:10" .*`},
+		{"setpriority 0-10", `cannot parse line: cannot parse token "0-10" .*`},
+		{"setpriority 0,1", `cannot parse line: cannot parse token "0,1" .*`},
+		{"setpriority 0x0", `cannot parse line: cannot parse token "0x0" .*`},
+		{"setpriority a1", `cannot parse line: cannot parse token "a1" .*`},
```

Comment:
You skipped `1a`. It was there since our syntax parser looks at the first character and the first character of `1a` is `1` which is valid, but `1a` should fail to parse.